app/controllers/statuses_controller.rb

```diff
@@ -64,6 +64,7 @@ def set_link_headers
   def set_status
     @status = @account.statuses.find(params[:id])
     authorize @status, :show?
+    raise Mastodon::NotPermittedError if current_account&.domain&.present? && @account.domain_blocking?(current_account.domain)
```

Comment:
This should just be part of `StatusPolicy` imo

Reply:
Could be, but it's really the only point where it can be called with a `current_account` that is of another domain

Reply:
It would be more clean and more future-proof. Also you're raising an exception just for it to be caught and re-raised as a different one straight away.